src/manifolds/StiefelCanonicalMetric.jl

```diff
@@ -0,0 +1,153 @@
+@doc raw"""
+    CanonicalMetric <: Metric
+
+The Canonical Metric is name used on several manifolds, for example for the [`Stiefel`](@ref)
+manifold, see for example[^EdelmanAriasSmith1998].
+
```

Comment:
If this is intended to be general, then perhaps this definition should be in `metric.jl` or another file? Then the reference should be moved to the docs page.

Reply:
Sure we can move it to the metric.jl file and docs. I am not 100% sure what you mean by moving the reference? It is shown in REPL this way and shows up on the docs page where it is used. Sure I would prefer a cite-style referencing, but that package/addon for Documenter is not yet ready.

Reply:
Hm, looking at it closely, I am not sure what it would be in the generic sense. Because canonical metric is usually always the euclidean one. So can we maybe only move it to a global case and move the reference for example to the exponential map, once we have a second canonical-non-Euclidean-metric case?